### DISCUSSION TITLE
v1.48.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#1818] [Announcements]
# February 2023 (version V1.48.0)

## 🔴 Breaking Changes and Important Notes for this release 🔴
- The server and the mobile app must be on the same version to function correctly.
- There are changes to the [`docker-compose.yml`](https://github.com/immich-app/immich/blob/4dab50c10a1cc196e973ed2b4751e8baebb89736/docker/docker-compose.yml#L34-L47) file for the `machine-learning` portion, and include the [`model-cache`](https://github.com/immich-app/immich/blob/4dab50c10a1cc196e973ed2b4751e8baebb89736/docker/docker-compose.yml#L93) volume at the end of the file as shown [here](https://github.com/immich-app/immich/blob/4dab50c10a1cc196e973ed2b4751e8baebb89736/docker/docker-compose.yml#L93). Please update yours accordingly.
- Temporarily dropping Machine Learning support for Arm64 and ArmV7 due to PyTorch not supporting these architectures from the Python stable wheels

## Highlights
Thanks to our talented contributors, I am excited to bring you to one of the biggest releases yet of Immich that I hope you will enjoy. This release brings a lot of new features and improvements to the app. Here are some of the highlights:
- **iOS Background Uploads**
- **Machine Learning moved to Python-based Image**
- **Auto-link Live Photos**

### iOS Background Uploads
I want to thank @martyfuhry for diving into this feature and his wife for wanting the feature :P. 

iOS users can now upload photos and videos in the background. This means that you can take photos and videos, which will be uploaded to your server without opening the app.
The background mechanism from the iPhone has two modes, a shorter task which runs around every 10 minutes and lasts for about 30 seconds, and a longer task which runs every 30-45 minutes and lasts longer. From our extensive testing, the app can upload about 1000 pictures overnight without plugging in and on WIFI.

> Please enable `Background App Refresh` in `Settings > General > Background App Refresh` for this feature to work.
> You can let the background uploads run on WIFI only or both WIFI and Cellular. This can be configured in the same menu in the abovementioned settings.

<p align="center">
<img src="https://user-images.githubusercontent.com/27055614/220419743-952d2de2-ed19-47fe-be11-53545170932f.png" width="250" title="background app refresh"> 

<img src="https://user-images.githubusercontent.com/27055614/220419705-110ae243-90b2-412c-8142-0183e5f79437.png" width="250" title="background app refresh"> 

</p>

You can get notified when the background task is running by enabling the notification setting from the Settings menu on Immich mobile app.

We are looking for feedback and bug reports on this feature. Please let us know if you have any issues with it.

### Machine Learning moved to Python-based Image
We are planning to revamp our search capability. The first step to making it smarter is to move our machine learning project from NodeJs based to Python based. This will allow us to use more powerful machine-learning libraries and improve the search.

We are planning to add facial recognition in the near future and also use [SBERT](https://www.sbert.net/examples/applications/image-search/README.html) for image search. This will allow you to search for images based on the content of the image.

![image](https://user-images.githubusercontent.com/27055614/220420047-48a4a97a-c958-4035-8d6d-f7cdf73c6f73.png)

This is a breaking change to the current setup. Please update your `docker-compose.yml` file for the `machine-learning` container accordingly. We are also dropping ML support for Arm64 and ArmV7 for now due to PyTorch not supporting these architectures from the Python stable wheels. We are looking into ways to support these architectures in the future.

### Auto-link Live Photos
LivePhotos’ motion and still parts are now automatically linked together. This means that you can upload two separate files from the web interface or from the CLI, and they will be linked into one LivePhoto asset and displayed correctly on the timeline.

For previously uploaded motion and picture parts of the LivePhoto asset from the web or the CLI, you can re-run the metadata extraction job to link them together.

----

And like other releases, we have made many other improvements and bug fixes for better performance and UI consistency across platforms. Please take a look at the full changelog below.

As always, thank you for using Immich, and I hope you enjoy this release.

Remember to support the project!

---- 
🎉 Cheer! 🎉

## Support

<p align="center">
<img src="https://media.giphy.com/media/l2YWkhC75gVi2TgCA/giphy.gif" width="250" title="supporting the project"> 
</p>

If you find the project helpful and it helps you in some ways, you can support the project [one time](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502) or [monthly](https://github.com/sponsors/alextran1502) from GitHub Sponsors

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

## Server
* feat(server): auto-link live photos by @jrasm91 in https://github.com/immich-app/immich/pull/1761
* refactor(server): remove checksum job by @jrasm91 in https://github.com/immich-app/immich/pull/1786
* infra(server): fix Album TypeORM relations and change ids to uuids by @zackpollard in https://github.com/immich-app/immich/pull/1582
* infra(server)!: fix typeorm asset entity relations by @zackpollard in https://github.com/immich-app/immich/pull/1782
* fix(server): use updated AssetEntity property in getting server info query by @alextran1502 in https://github.com/immich-app/immich/pull/1797
* chore(server): typeorm definitions fix part 3 by @zackpollard in https://github.com/immich-app/immich/pull/1796
* fix(server): asset search query by @michelheusschen in https://github.com/immich-app/immich/pull/1806
* fix(server): use fileCreatedAt for asset sorting after recent refactor by @zackpollard in https://github.com/immich-app/immich/pull/1799
* feat(server)!: move machine learning to Python based image by @alextran1502 in https://github.com/immich-app/immich/pull/1774

## Mobile
* feat(mobile): Add Swedish i18n by @MrYakobo in https://github.com/immich-app/immich/pull/1769
* feat(mobile): Scroll to top when tapping photos while already on photo page by @martyfuhry in https://github.com/immich-app/immich/pull/1784
* fix(mobile): Delete goes to next page instead of popping back to the main timeline by @martyfuhry in https://github.com/immich-app/immich/pull/1781
* feat(mobile): iOS background sync by @martyfuhry in https://github.com/immich-app/immich/pull/1758
* feat(mobile): added more translations by @michelheusschen in https://github.com/immich-app/immich/pull/1805
* feat(mobile): improve date formatting by @michelheusschen in https://github.com/immich-app/immich/pull/1804
* chore(mobile): styling and linter by @alextran1502 in https://github.com/immich-app/immich/pull/1807
* fix(mobile): user get logged out upon clicking on any thing after logging in by @alextran1502 in https://github.com/immich-app/immich/pull/1808
* feat(mobile): iOS background sync notifications by @martyfuhry in https://github.com/immich-app/immich/pull/1811
* chore(mobile): Improve reliability of asset loading and indexing by @alextran1502 in https://github.com/immich-app/immich/pull/1813
* fix(mobile): Fixed load translations in background sync by @martyfuhry in https://github.com/immich-app/immich/pull/1810

## Web
* refactor(web): use cookies interface by @michelheusschen in https://github.com/immich-app/immich/pull/1777
* feat(web): only show copy image when supported by @michelheusschen in https://github.com/immich-app/immich/pull/1776
* fix(web): user profile 404 by @jrasm91 in https://github.com/immich-app/immich/pull/1800
* feat(web): add some material design 3 styling to forms and ui elements by @alextran1502 in https://github.com/immich-app/immich/pull/1798
* fix(web): loading profile image by @michelheusschen in https://github.com/immich-app/immich/pull/1803
* feat(web): styling sidebar buttons by @michelheusschen in https://github.com/immich-app/immich/pull/1812
* fix(web): modal overlay hidden behind nav bar by @alextran1502 in https://github.com/immich-app/immich/pull/1816

## Others
* fix(release): pump script by @jrasm91 in https://github.com/immich-app/immich/pull/1768
* fix(docs): Tiny tweaks in unraid guide by @bo0tzz in https://github.com/immich-app/immich/pull/1772
* fix(ci): Repair broken docker build matrix by @bo0tzz in https://github.com/immich-app/immich/pull/1793
* chore(ci): increase mobile integration test speed and reliability by @zackpollard in https://github.com/immich-app/immich/pull/1792

## New Contributors
* @MrYakobo made their first contribution in https://github.com/immich-app/immich/pull/1769

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.47.3...v1.48

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.48.0'>v1.48.0</a>.</em>

**Comment by Dleewee** [comment]
Hi, 
If anyone else is seeing "undefined volume model-cache: invalid compose project" you need to add model-cache: under the volumes at the bottom of the compose file. The release yml file is not showing that as a change.

**Reply by alextran1502** [reply]
Hmm, I just checked and it is there. 🤔

**Reply by Dleewee** [reply]
I do see it as well, but it was not being highlighted as a change on the release commit (when I clicked on the docker-compose.yml link within the release notes).
No worries, I just wanted to help others who are upgrading and were specifically looking for changed lines.

**Reply by alextran1502** [reply]
Ah I see, thank you. I cannot highlight 2 separated sections in the same link it seems like.

**Comment by bertmelis** [comment]
I suppose the machine learning image for old CPU's isn't needed anymore.

I can't look into the update for the next week. Are there any compatibility issues to be expected (except for the arm CPU's as written)?

**Reply by alextran1502** [reply]
Hi @bertmelis thank you for maintaining the previous image. It is no longer needed. I believe the current PyTorch image supports non-avx, I haven't tested yet but from some discussion it seems to be working.

**Reply by bertmelis** [reply]
Meanwhile, I'm running the new machine learning on my J4105 processor. No problems.

Time to deprecate the alt image... I give it some time though and put it to sleep gently.

**Reply by Synoxion** [reply]
I tested it on my Pentium G4560 and yes, it works.
